lib/python/pyflyby/_cmdline.py

```diff
@@ -380,14 +384,35 @@ def __del__(self):
 
 
 def process_actions(filenames:List[str], actions, modify_function,
-                    reraise_exceptions=()):
+                    reraise_exceptions=(), exclude=()):
+
+    if not isinstance(exclude, (list, tuple)):
+        raise ConfigurationError(
+            "Exclusions must be a list of filenames/patterns to exclude."
+        )
+
     errors = []
     def on_error_filename_arg(arg):
         print("%s: bad filename %s" % (sys.argv[0], arg), file=sys.stderr)
         errors.append("%s: bad filename" % (arg,))
-    filenames = filename_args(filenames, on_error=on_error_filename_arg)
+    filename_objs = filename_args(filenames, on_error=on_error_filename_arg)
     exit_code = 0
-    for filename in filenames:
+    for filename in filename_objs:
+
+        # Log any matching exclusion patterns before ignoring, if applicable
+        matching_excludes = []
+        for pattern in exclude:
+            if Path(str(filename)).match(str(pattern)):
```

Comment:
I think we should consider if it is better to use `Path.match` or regular expressions. I think `Path.match` is more convenient but `black` uses regular expressions so it would introduce some friction for users how would like to use the same exclude pattern for both `black` and `pyflyby`. On the other hand, more modern tooling like `ruff` users glob patterns for `--exclude`.